### DISCUSSION TITLE
FastAPI: Internal server error but still able to delete and post new data

### DISCUSSION THREAD

**Discussion by Nathan-Bernardo** [fastapi/fastapi#8923] [Questions] [answered]
Hi everyone,

I'm fairly new to FastAPI.  I am creating a CRUD app that is supposed to store a candidate's information into MongoDB.  What is going right now is that I'm getting an Internal Server Error whenever I try to update, delete, post, or get data  (shown in the picture).  However, I am still able to delete and post new data to MongoDB with the Internal Server Error.  
![image](https://user-images.githubusercontent.com/86064236/124227993-5afc6b00-dac0-11eb-8eff-9675286f33ea.png)

Here is what my code looks like: https://gist.github.com/Nathan-Bernardo/a4d0e4c5928896fefa2463f9d9c8e483.   What seems to be the problem here and how can I fix it?  Thank you!

**Comment by Mause** [comment]
Is an error being logged in the console?

**Comment by xxl1212** [comment]
Can you show me the console where you launched your application?

**Comment by alfiephillips** [comment]
The error can't be related to MongoDB since the database requests are still going through, the error can be something to do with how your sending the data, it may be in the wrong format that the server can't handle. Please could you show the error?

**Comment by Nathan-Bernardo** [comment]
This is the error I get on the console 
`INFO:     127.0.0.1:43662 - "DELETE /candidates/64516637-0019-4b1c-8b96-b58bc22e48e4 HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
.
.
.
File "/home/ncbernar/Github/listeedemo/backend/app/ResumeParser/routers.py", line 37, in delete_candidate
    delete_candidate = await request.app.mongodb["candidates"].delete_one({"_id": id})
TypeError: object DeleteResult can't be used in 'await' expression `

**Comment by alfiephillips** [comment]
> DeleteResult can't be used in 'await' expression

Have you made sure this is executed in an asynchronous function? Or the function your using supports promises? Search the documentation up, and you shall find your solution.

**Comment by Mause** [comment, marked as answer]
Well there's your issue right off - you're awaiting something that isn't awaitable
